Orange/data/table/base.py

```diff
@@ -0,0 +1,1383 @@
+            result[icn] = first_col
+
+        # process a list of variables, appending suffix if not found in the
+        # resulting columns (that means it was a dup)
+        def suffix_dups(varlist, suffix):
+            return [v if v in result.columns else v.copy(new_name=v.name + suffix) for v in varlist]
+
+        # dedup because the target join valriable doesn't get renamed and there is
+        # only one column, wehile without this, the domain would have two
+        def dedup_inorder(varlist):
+            s = set()
+            return [x for x in varlist if not (x in s or s.add(x))]
+
+        # merge domain
+        new_domain = Domain(
+            dedup_inorder(suffix_dups(self.domain.attributes, suffix_left) +
+                          suffix_dups(right.domain.attributes, suffix_right)),
+            dedup_inorder(suffix_dups(self.domain.class_vars, suffix_left) +
+                          suffix_dups(right.domain.class_vars, suffix_right)),
+            dedup_inorder(suffix_dups(self.domain.metas, suffix_left) +
+                          suffix_dups(right.domain.metas, suffix_right))
+        )
+        result.domain = new_domain
+
+        # always reindex
+        result.index = self._new_id(len(result), force_list=True)
+
+        return result
+
+    def approx_len(self):
+        """Return the approximate length of the table."""
+        # TODO: remove, here solely for SQL compatibility, which would ideally be replaced with Spark
+        # https://github.com/biolab/orange3/wiki/pandas-migration#future-todos-and-architectural-wishlist
+        return len(self)
+
+    def exact_len(self):
+        """Return the exact length of the table."""
+        # TODO: remove, here solely for SQL compatibility, which would ideally be replaced with Spark
+        # https://github.com/biolab/orange3/wiki/pandas-migration#future-todos-and-architectural-wishlist
+        return len(self)
+
+    def has_missing(self):
+        """Return `True` if there are any missing attribute or class values."""
+        # manual access to columns because dumping to a numpy array (with self.X) is slower
+        return self[self.domain.attributes].isnull().any().any() or self.has_missing_class()
+
+    def has_missing_class(self):
+        """Return `True` if there are any missing class values."""
+        return self[self.domain.class_vars].isnull().any().any()
+
+    def iterrows(self):
+        """An override to return TableSeries instead of Series (pandas doesn't do that by default)."""
+        # TODO: remove this in next pandas release, when it works by default (pydata/pandas#13978)
+        # super here is the next item in the MRO, e.g. pd.DataFrame or pd.SparseDataFrame
+        gen = super().iterrows()
+        for item in gen:
+            yield (item[0], self._constructor_sliced(item[1]))
+
+    def __hash__(self):
+        # TODO: inconsistent when dtype=object
+        return hash(bytes(self.values))
```

Comment:
```py
return hash(bytes(self.values if include_metas else self._to_numpy(X=True, Y=True)))
```
Nobody ever used `include_metas=True`, so it's up to you to design the new interface anyway.